File: dart/hamming/lib/hamming.dart
class Hamming {
  int distance(String s1, String s2) {
    if (s1.length != s2.length) {
      throw ArgumentError('Strands must be the same length');
    }
    int count = 0;
    for (int i = 0; i < s1.length; i++) {
      if (s1[i] != s2[i]) {
        count++;
      }
    }
    return count;
  }
}

File: csharp/hamming/Hamming.cs
using System;
using System.Linq;


public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        if (firstStrand.Length != secondStrand.Length) {
            throw new ArgumentException("Strands must be the same length.");
        }
        // return firstStrand.Select((c, i) => c == secondStrand[i] ? 0 : 1).Sum();
        return firstStrand.Zip(secondStrand, (first, second) => first == second ? 0 : 1).Sum();
    }
}